code/_helpers/unsorted.dm

```diff
@@ -589,62 +589,149 @@ Turf and target are seperate in case you want to teleport some distance from a t
 	if (progbar)
 		qdel(progbar)
 
-/proc/do_after(mob/user as mob, delay as num, needhand = TRUE, atom/movable/act_target = null, use_user_turf = FALSE, display_progress = TRUE, datum/callback/extra_checks)
-	if(!user || isnull(user))
-		return 0
+/// Integer. Unique sequential ID from the `do_after` proc used to validate `DO_USER_UNIQUE_ACT` flag checks.
+/mob/var/do_unique_user_handle = 0
+/// The mob currently interacting with the atom during a `do_after` timer. Used to validate `DO_TARGET_UNIQUE_ACT` flag checks.
+/atom/var/do_unique_target_user
+
+/**
+ * Timed actions involving one mob user and (optionally) one target.
+ *
+ * Returns TRUE on success or FALSE on failure
+ *
+ * Arguments:
+ * * user: The user to check for.
+ * * delay: The delay in ticks to wait before returning TRUE.
+ * * target: The target to check for. Optional.
+ * * do_flags: Flags that determine what the user and target can and cannot do, defined in [mobs.dm]. Defaults to DO_DEFAULT.
+ * * incapacitation_flags: Incapacitation flags that determines if the user can be incapacitated. Defaults to INCAPACITATION_DEFAULT.
+ * * extra_checks: Optional extra checks, that uses a callback. See [datum/callback].
+ *
+ */
+/proc/do_after(mob/user, delay, atom/target, do_flags = DO_DEFAULT, incapacitation_flags = INCAPACITATION_DEFAULT, datum/callback/extra_checks)
+	return !do_after_detailed(user, delay, target, do_flags, incapacitation_flags)
```

Comment:
The point of `do_after_detailed` is to return the exact error as a flag, for custom detailed error handling. But this would have an opposite effect of what `do_after` does on other servers, so we just negate the result here so `0` would return as TRUE, while any non-zero value (any error) would return as FALSE

Reply:
Why not have it as a parameter/flag then?

You could have eg. a `return_detailed_error` boolean at the end of the function header, set to FALSE by default, and at the end of the function decide based on that decide if to return the error flag or the simplified TRUE/FALSE